### PR TITLE
min / max for lat long values

### PR DESCRIPTION
**Pull request by actuallymab**
Sometimes you may want to set lat/lng parameters for a spesific area. 

Ex: 
For Istanbul:
'lat' => $faker->latitude(40.958836, 41.136456),
'lng' => $faker->longitude(28.64, 29.98),

### CODE REVIEW

File: src/Faker/Provider/Address.php [hunk 1]
```diff
@@ -104,20 +104,24 @@ public static function country()
     }
 
     /**
-     * @example 77.147489
-     * @return float Uses signed degrees format (returns a float number between -90 and 90)
+     * @example '77.147489'
+     * @param int $min
```
Comment (fzaninotto) [inline src/Faker/Provider/Address.php]: it's a `float`, not an `int`, right?

File: src/Faker/Provider/Address.php [hunk 2]
```diff
@@ -104,20 +104,24 @@ public static function country()
...
+     * @param int $max
+     * @return string
      */
-    public static function latitude()
+    public static function latitude($min = -90, $max = 90)
     {
-        return static::randomFloat(6, 0, 180) - 90;
+        return number_format(mt_rand($min * 1000000, $max * 1000000)/1000000, 6);
```
Comment (fzaninotto) [inline src/Faker/Provider/Address.php]: why don't you use `random_float` anymore?

**Comment by fzaninotto** [comment]
Please also update the README to add an example with the new parameter

File: src/Faker/Provider/Address.php (hunk 2 above)
Comment (aydnbhdr) [inline src/Faker/Provider/Address.php]: this is a coordinate, not a "just" float number. So I decided to update lat lng functions

File: src/Faker/Provider/Address.php (hunk 1 above)
Comment (aydnbhdr) [inline src/Faker/Provider/Address.php]: Sorry for missing that, Phpstorm decided it, not me :)

File: src/Faker/Provider/Address.php [hunk 3]
```diff
@@ -104,20 +104,24 @@ public static function country()
...
+     * @example '77.147489'
+     * @param float|int $min
+     * @param float|int $max
      * @return float Uses signed degrees format (returns a float number between -90 and 90)
      */
-    public static function latitude()
+    public static function latitude($min = -90, $max = 90)
     {
-        return static::randomFloat(6, 0, 180) - 90;
+        return floatval(number_format(mt_rand($min * 1000000, $max * 1000000)/1000000, 6));
```
Comment (fzaninotto) [inline src/Faker/Provider/Address.php]: I don't understand the interest of converting a float to string (with `number_format`) and then back to float (with `floatval`)

**Comment by fzaninotto** [comment]
Tests fail

File: src/Faker/Provider/Address.php (hunk 2 above)
Comment (fzaninotto) [inline src/Faker/Provider/Address.php]: Still, `static::randomFloat()` seems to be a better option:

```php
// not that clear
return mt_rand($min * 1000000, $max * 1000000)/1000000;
// better
return static::randomFloat(6, $min, $max);
```

**Comment by fzaninotto** [comment]
Also, please see #416 for why you should use `randomFloat()`

**Comment by ikwattro** [comment]
@fzaninotto @actuallymab Any update on this PR, we are really interested in this feature

**Comment by actuallymab** [comment]
don't know.. @fzaninotto ok with this?

File: src/Faker/Provider/Address.php [hunk 4]
```diff
@@ -104,20 +104,24 @@ public static function country()
...
 
     /**
-     * @example 77.147489
+     * @example '77.147489'
+     * @param float|int $min
+     * @param float|int $max
      * @return float Uses signed degrees format (returns a float number between -90 and 90)
      */
-    public static function latitude()
+    public static function latitude($min = -90, $max = 90)
```
Comment (fzaninotto) [inline src/Faker/Provider/Address.php]: You're changing the results of existing calls. Please use 0 and 180 as min and max values.

File: src/Faker/Provider/Address.php [hunk 5]
```diff
@@ -104,20 +104,24 @@ public static function country()
...
 
     /**
-     * @example 86.211205
+     * @example '86.211205'
+     * @param float|int $min
+     * @param float|int $max
      * @return float Uses signed degrees format (returns a float number between -180 and 180)
      */
-    public static function longitude()
+    public static function longitude($min = -180, $max = 180)
```
Comment (fzaninotto) [inline src/Faker/Provider/Address.php]: You're changing the results of existing calls. Please use 0 and 360 as min and max values.

File: src/Faker/Provider/me_ME/Address.php [hunk 6]
```diff
@@ -111,17 +111,23 @@ public function cityName()
 
     /**
      * @example '77.147489'
+     * @param float|int $min
+     * @param float|int $max
+     * @return float Uses signed degrees format (returns a float number between -90 and 90)
      */
-    public static function latitude()
+    public static function latitude($min = 42.43, $max = 42.45)
```
Comment (fzaninotto) [inline src/Faker/Provider/me_ME/Address.php]: `latitude()` and `longitude()` shouldn't be localized in the first place. This is a mistake in the existing `me_ME` formatters. Please create a new formatter, like `localGeolocation()`, for that purpose. That way, you can generate latite and longitude within a polygon.

File: src/Faker/Provider/Address.php (hunk 4 above)
Comment (TimWolla) [inline src/Faker/Provider/Address.php]: He isn't. Look again.

Comment (fzaninotto) [inline src/Faker/Provider/Address.php]: My bad, you're right.

File: src/Faker/Provider/Address.php (hunk 5 above)
Comment (fzaninotto) [inline src/Faker/Provider/Address.php]: Disregard.

File: src/Faker/Provider/me_ME/Address.php (hunk 6 above)
Comment (fzaninotto) [inline src/Faker/Provider/me_ME/Address.php]: This is still an issue.

**Comment by fzaninotto** [comment]
Thanks

**Pull request merged**